crates/sui-core/src/transaction_manager.rs

```diff
@@ -30,14 +30,18 @@ pub struct TransactionManager {
 
 #[derive(Default)]
 struct Inner {
+    // Current epoch of TransactionManager.
+    epoch: EpochId,
```

Comment:
why not just put Arc of epoch store?

Reply:
Ah I guess epoch is better since there is no guarantee that transaction manager stops at epoch boundary, so we need to check on epoch equality each time using it

Reply:
Only epoch ID would be needed, especially if we want to go with https://github.com/MystenLabs/sui/pull/7487. Alternatively, we can remove the epoch store args. I don't have preference either way.